lib/eslint.js

```diff
@@ -908,6 +908,8 @@ module.exports = (function() {
                 }
             }
 
+            ast.hasShebang = !!shebang;
```

Comment:
Here's the line where the `SourceCode` instance is created: https://github.com/eslint/eslint/blob/master/lib/eslint.js#L817

And here's where the shebang comment is removed: 
https://github.com/eslint/eslint/blob/master/lib/eslint.js#L903